obsreport/obsreport_scraper.go

```diff
@@ -48,15 +66,58 @@ type ScraperSettings struct {
 
 // NewScraper creates a new Scraper.
 func NewScraper(cfg ScraperSettings) *Scraper {
-	return &Scraper{
+	return newScraper(cfg, featuregate.GetRegistry())
+}
+
+func newScraper(cfg ScraperSettings, registry *featuregate.Registry) *Scraper {
+	scraper := &Scraper{
 		level:      cfg.ReceiverCreateSettings.TelemetrySettings.MetricsLevel,
 		receiverID: cfg.ReceiverID,
 		scraper:    cfg.Scraper,
 		mutators: []tag.Mutator{
 			tag.Upsert(obsmetrics.TagKeyReceiver, cfg.ReceiverID.String(), tag.WithTTL(tag.TTLNoPropagation)),
 			tag.Upsert(obsmetrics.TagKeyScraper, cfg.Scraper.String(), tag.WithTTL(tag.TTLNoPropagation))},
 		tracer: cfg.ReceiverCreateSettings.TracerProvider.Tracer(cfg.Scraper.String()),
+
+		logger:            cfg.ReceiverCreateSettings.Logger,
+		useOtelForMetrics: registry.IsEnabled(obsreportconfig.UseOtelForInternalMetricsfeatureGateID),
+		otelAttrs: []attribute.KeyValue{
+			attribute.String(obsmetrics.ReceiverKey, cfg.ReceiverID.String()),
+			attribute.String(obsmetrics.ScraperKey, cfg.Scraper.String()),
+		},
+	}
+
+	scraper.createOtelMetrics(cfg)
+	return scraper
+}
+
+func (s *Scraper) createOtelMetrics(cfg ScraperSettings) {
+	meter := cfg.ReceiverCreateSettings.MeterProvider.Meter(scraperScope)
```

Comment:
Move this meter creation to be after the `if` bellow.

Reply:
Thanks for the review! Fixed.